code/game/gamemodes/marker/gamemodes.dm

```diff
@@ -123,4 +118,29 @@ GLOBAL_DATUM_INIT(shipsystem, /datum/ship_subsystems, new)
 	if (GM.marker_active)
 		to_chat(src, "The marker is already active")
 		return
-	GM.activate_marker()
\ No newline at end of file
+	GM.activate_marker()
+
+/**
+
+There is probably a better way to do this, but I've added some more stringent checks to the gamemode win conditions to prevent things like:
+Admin characters being counted as crew, and on their deletion, ending the game
+Non-critical characters like any ghost-roles you may wish to add, or even antags, from counting as "dead crew" (Antags fallback to their own "are the antags dead checks"
+
+*/
+
+//Handlers to end the game if everybody dies.
+/mob/living/carbon/human/death(gibbed,deathmessage="seizes up and falls limp...", show_dead_message = "You have died.")
+	. = ..()
+	var/datum/computer_file/report/crew_record/R = get_crewmember_record(real_name) //Try get a crew manifest for this mob
+	if(!R)
+		return FALSE //Not on the manifest? You don't exist.
+	if(istype(ticker.mode, /datum/game_mode/marker)) //Precondition: Ensure that the game mode is one of ours, we don't want to say, override traitor or anything.
+		ticker.mode.check_finished()
+
+//Method to be called when a human player dies for any reason, this will check the amount of players that should be alive, and sees if they're dead or not. We're doing it via an almost signal (bay lacks signals) to avoid processing overhead.
+/datum/game_mode/marker/check_finished()
+	var/all_mobs = GLOB.all_crew_records.len //If you're not in the manifest I don't care about your existence. Updates this every time we check a win to ensure that no new players have joined the fray.
```

Comment:
Put a check here to see if the marker has activated yet. We don't want the round to end because one player spawned in alone, and shot themselves. This victory condition should only be able to activate after the necromorphs come into play